src/renderers/WebGLRenderer.js

```diff
@@ -49,6 +49,10 @@ THREE.WebGLRenderer = function ( parameters ) {
 
 	this.sortObjects = true;
 
+	// user-defined clipping
+
+	this.userClippingPlanes = [];
```

Comment:
As far as I understand, this is mainly uniforms. Wouldn't it be better to set this array in the material instead?